packages/@aws-cdk/aws-ecs-patterns/lib/base/application-load-balanced-service-base.ts

```diff
@@ -210,48 +195,27 @@ export abstract class LoadBalancedServiceBase extends cdk.Construct {
     this.assignPublicIp = props.publicTasks !== undefined ? props.publicTasks : false;
     this.desiredCount = props.desiredCount || 1;
 
-    // Load balancer
-    this.loadBalancerType = props.loadBalancerType !== undefined ? props.loadBalancerType : LoadBalancerType.APPLICATION;
-
-    if (this.loadBalancerType !== LoadBalancerType.APPLICATION && this.loadBalancerType !== LoadBalancerType.NETWORK) {
-       throw new Error(`invalid loadBalancerType`);
-    }
-
     const internetFacing = props.publicLoadBalancer !== undefined ? props.publicLoadBalancer : true;
 
     const lbProps = {
       vpc: this.cluster.vpc,
       internetFacing
     };
 
-    if (this.loadBalancerType === LoadBalancerType.APPLICATION) {
-      this.loadBalancer = new ApplicationLoadBalancer(this, 'LB', lbProps);
-    } else {
-      this.loadBalancer = new NetworkLoadBalancer(this, 'LB', lbProps);
-    }
+    this.loadBalancer = new ApplicationLoadBalancer(this, 'LB', lbProps);
 
     const targetProps = {
       port: 80
     };
 
-    const hasCertificate = props.certificate !== undefined;
-    if (hasCertificate && this.loadBalancerType !== LoadBalancerType.APPLICATION) {
-      throw new Error("Cannot add certificate to an NLB");
-    }
-
-    if (this.loadBalancerType === LoadBalancerType.APPLICATION) {
-      this.listener = (this.loadBalancer as ApplicationLoadBalancer).addListener('PublicListener', {
-        port: hasCertificate ? 443 : 80,
-        open: true
-      });
-      this.targetGroup = this.listener.addTargets('ECS', targetProps);
+    this.listener = this.loadBalancer.addListener('PublicListener', {
+      port: props.certificate !== undefined ? 443 : 80,
```

Comment:
See above

Reply:
As discussed made TaskDefinition a property of the construct

Reply:
This isn't a field on the Task Definition though -- it's on the listener.